syntax = "proto3";

package privacy;

// Esta linha é obrigatória para o gerador de código Go
option go_package = "backend-go/pb";

service PrivacyService {
  rpc ProcessDataset (AnonymizeRequest) returns (AnonymizeResponse) {}
}

message AnonymizeRequest {
  string input_path = 1;
  float epsilon = 2;
  float delta = 3;
  bool detect_pii = 4;
}

message AnonymizeResponse {
    string output_path = 1;
    float privacy_score = 2;   // Este será o 1 - max(riscos)
    float utility_score = 3;
    float epsilon_used = 4;
    string status = 5;         // Aqui enviaremos a tabela formatada
    map<string, string> pii_report = 6;    
    // Novos campos para a GUI tabular
    float singling_out_risk = 7;
    float linkability_risk = 8;
    float inference_risk = 9;
}